ecgtools/parsers.py

```diff
@@ -92,3 +100,252 @@ def cmip6_default_parser(
         data = {'exception': str(e), 'file': filepath}
         print(data)
         return {}
+
+
+class YAML_Parser:
+    """
```

Comment:
Can we merge some of the functionality in this class into the main `Builder` class, and keep the remaining functionality (such as the `parser()` and `_parser_netcdf()` methods) as simple functions that integrates well with the `Builder` class?

Reply:
@andersy005 possibly.  What is left is the yaml parsing code.  I went this route and overloaded "parsers" to include parsing the yaml syntax.  I then included it all into the same class for usability reasons.  I didn't really see the yaml parsing fitting into the `Builder` class based on the functionality that exists within it to build a catalog.  Should it exist within a tools.py file instead?  Or do you think it still fits within the `Builder` class?

Reply:
@andersy005:  What parts do you think belong in `Builder`?

Reply:
In conversation with @kmpaul and @andersy005 , we decided to leave this code within `parser.py`.